lab_2_retrieval_w_bm25/main.py

```diff
@@ -18,7 +19,21 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
-
+    if not isinstance(text, str):
+        return None
+    new_text = ''
+    tokenized_list = []
+    for sign in text.lower():
+        if not sign.isalpha():
+            added = ' '
+        else:
+            added = sign
+        new_text = f'{new_text}{added}'
+    while '  ' in new_text:
+        new_text = new_text.replace('  ', ' ')
+    for token in new_text.split():
+        tokenized_list.append(token)
+    return tokenized_list
```

Comment:
this is very difficult to understand, simplify

Reply:
Done!